sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
+
+### DefaultAzureCredential
+
+`DefaultAzureCredential` simplifies authentication while developing apps that deploy to Azure by combining credentials used in Azure hosting environments with credentials used in local development. For more information, see [DefaultAzureCredential overview][dac_overview].
+
+#### Continuation policy
+
+As of version 1.14.0<!-- TODO: I'm not sure what version in the rust libraries this statement becomes true at. My guess would be at our first official beta version. Maybe in that case we can just remove the version specification since this will always be true when using the official library. -->, `DefaultAzureCredential` attempts to authenticate with all developer credentials until one succeeds, regardless of any errors previous developer credentials experienced. For example, a developer credential may attempt to get a token and fail, so `DefaultAzureCredential` will continue to the next credential in the flow. Deployed service credentials stop the flow with a thrown exception if they're able to attempt token retrieval, but don't receive one. Prior to version 1.14.0, developer credentials would similarly stop the authentication flow if token retrieval failed, but this is no longer the case.
+
+This allows for trying all of the developer credentials on your machine while having predictable deployed behavior.
+
+## Examples
+
+The following examples are provided:
+<!-- no toc -->
+* [Authenticate with DefaultAzureCredential](#authenticate-with-defaultazurecredential "Authenticate with DefaultAzureCredential")
+<!-- TODO: Un-Comment if/when the sections are added>
+* [Define a custom authentication flow with ChainedTokenCredential](#define-a-custom-authentication-flow-with-chainedtokencredential "Define a custom authentication flow with ChainedTokenCredential")
+* [Async credentials](#async-credentials "Async credentials")
+</!-->
+
+### Authenticate with `DefaultAzureCredential`
+
+More details on configuring your environment to use `DefaultAzureCredential` can be found in the class's [reference documentation][default_cred_ref].
+
+This example demonstrates authenticating the `CosmosClient` from the [azure_data_cosmos][azure_data_cosmos] library using `DefaultAzureCredential`.
 
 ```rust
-use azure_core::credentials::TokenCredential;
+use azure_data_cosmos::CosmosClient;
 use azure_identity::DefaultAzureCredential;
-# use azure_core::{ClientOptions, Result};
-# use std::sync::Arc;
-# struct StorageAccountClient;
-# impl StorageAccountClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(StorageAccountClient))
-#     }
-# }
-# struct SecretClient;
-# impl SecretClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(SecretClient))
-#     }
-# }
 
 let credential = DefaultAzureCredential::new().unwrap();
```

Comment:
In our READMEs, I think we should show best practices as much as possible. But these READMEs also need to be compiled as documented: https://azure.github.io/azure-sdk/rust_introduction.html#rust-documentation-readme (or maybe that was already in there).

Thus, this whole section might actually be:

```rust
use azure_data_cosmos::CosmosClient;
use azure_identity::DefaultAzureCredential;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let credential = DefaultAzureCredential::new()?;
    let client = CosmosClient::new("https://myaccount.documents.azure.com/", credential.clone(), None)?;
    // ...
}
```

Reply:
Updated example